WHID_Elite_HybridCC_v.1.0/WHID_Elite_HybridCC_v.1.0.ino

```diff
@@ -0,0 +1,463 @@
+// ### hybridC2off:                  ### Example: ###   hybridC2off:                                 #
+// ###################################################################################################
+
+#include "Adafruit_FONA.h"
+#include "FingerprintUSBHost.h"
+#include <Keyboard.h>
+
+int dontSMS=0; // Set to 1 to save SMS
+int dontGPRS=0; // Set to 1 to use GPRS
+char gprsAirgap[50];
+char fonaNotificationBuffer[64];
+char smsBuffer[150]; //Default 250. If changed, need also to change Line 615 accordingly.
+ 
+// Dec to Bin https://forum.arduino.cc/index.php?topic=412553.0
+char *tobin32(unsigned long x)
+{
+  static char b[33]; //this size should be enough for any ASK/OOK protocol out there. If not, change it accordingly.
+  b[32] = '\0';
+
+  for ( int z = 0; z < 32; z++) {
+    b[31 - z] = ((x >> z) & 0x1) ? '1' : '0';
+  }
+
+  return b;
+}
+
+#define FONA_RX 2
+#define FONA_TX 3
+#define FONA_RST 4
+
+HardwareSerial *fonaSerial = &Serial1;
+Adafruit_FONA fona = Adafruit_FONA(FONA_RST);
+
+//This function return the amount of free memory available. Is used only for debugging.
+int freeRam () {
+  extern int __heap_start, *__brkval;
+  int v;
+  return (int) &v - (__brkval == 0 ? (int) &__heap_start : (int) __brkval);
+}
+
+void typeKey(uint8_t key) {
+  Keyboard.press(key);
+  delay(50);
+  Keyboard.release(key);
+}
+
+void setup() {
+  Keyboard.begin();
+
+  // ------ Initialize SIM800 ------
+  pinMode(13, OUTPUT);
+  digitalWrite(13, LOW); //(SIM800 gets resetted ONLY when RST is pulled LOW)
+  delay(50);
+  digitalWrite(13, HIGH);
+  pinMode(2, OUTPUT);
+  digitalWrite(2, HIGH); //HIGH == connected to GND
+  delay(1000);
+  digitalWrite(2, LOW); 
+  // ------ Initialize SIM800 ------
+
+
+  // #########################################################################################
+  // ##### This must be commented. Otherwise the script will never start automatically. ######
+  // ##### Uncomment, only if you wanna use serial monitor to debug the whole workflow  ######
+  // ##### and you wanna start SIM800 manually.                                         ######
+  // #########################################################################################
+  //while (!Serial);
+  // #########################################################################################
+
+  Serial.begin(4800);
+  fonaSerial->begin(4800);
+  if (! fona.begin(*fonaSerial)) {
+    Serial.println(F("NO SIM800!"));
+    while (1);
+  }
+  fona.unlockSIM("1234"); // ### Change this to your SIM's Pincode (if any) ###
+  //fonaSerial->print("AT+CLCK=\"SC\",0,\"1234\"\r\n"); // ### This remove permanently your SIM's Pincode ###
+  
+  //delay(3000); //###LBO: Original Working Delay
+  delay(1000);
+  fonaSerial->print("AT+CNMI=2,1\r\n");  //Set up the SIM800 to send a +CMTI notification when an SMS is received
+  delay(500);
+  fona.setGPRSNetworkSettings(F("internet")); //### APN Setting in case you MNO requires specific ones ###
```

Comment:
Here is were I use the FONA library to set the APN settings:
//fona.setGPRSNetworkSettings(F("your APN"), F("your username"), F("your password"));